### ISSUE TITLE
logging TODO

### ISSUE THREAD

**Issue by hannesm** [mirage/mirage#754]
via #749:
split mirage-logs into two pieces:
- the ring buffer (and a `--debug-on-crash` command line flag)
- a console reporter

and then handle syslog vs reporters in a same way, and respect the ring buffer filling on the syslog code path

**Comment by hannesm** [comment]
some other problems with current `mirage-logs`:
- while Logs uses `warn` as default logging level, mirage-logs uses `info`
- disabling logging can be achieved via the `no_reporter`, but this has to be passed as an argument to `register` in `config.ml`.  A command-line switch to mirage configure achieving the same thing (no logging, use console, use syslog) without having to modify `config.ml` would be appreciated.

**Comment by hannesm** [comment]
I'll close this issue. The logging is fine.

Indeed, a "standard" unikernel utilizes `mirage-logs`, which verbosity can be configured with `-l` command line flag. Syslog can be used, but needs more external libraries (and not every unikernel should require syslog).

FWIW, how I currently configure "optional syslog" is:

(a) define a "--enable-monitoring" configure-time argument
(b) and a mangement stack (i.e. second network device)
(c) define a "--name`
(d) define a syslog argument
(e) run a separate job that logs

```OCaml
let enable_monitoring =
  let doc = Key.Arg.info
      ~doc:"Enable monitoring (only available for solo5 targets)"
      [ "enable-monitoring" ]
  in
  Key.(create "enable-monitoring" Arg.(flag ~stage:`Configure doc))

let stack = generic_stackv4v6 default_network

let management_stack =
  if_impl
    (Key.value enable_monitoring)
    (generic_stackv4v6 ~group:"management" (netif ~group:"management" "management"))
    stack

let name =
  let doc = Key.Arg.info ~doc:"Name of the unikernel" [ "name" ] in
  Key.(v (create "name" Arg.(opt string "robur.coop" doc)))

let syslog =
  let syslog =
    let doc = Key.Arg.info ~doc:"syslog host IP" ["syslog"] in
    Key.(v (create "syslog" Arg.(opt (some ip_address) None doc)))
  in
  let connect _ modname = function
    | [ _ ; stack ] ->
      Fmt.str "Lwt.return (match %a with\
               | None -> Logs.warn (fun m -> m \"no syslog specified, dumping on stdout\")\
               | Some ip -> Logs.set_reporter (%s.create %s ip ~hostname:%a ()))"
        Key.serialize_call syslog modname stack
        Key.serialize_call name
    | _ -> assert false
  in
  impl
    ~packages:[ package ~sublibs:["mirage"] ~min:"0.4.0" "logs-syslog" ]
    ~keys:[ name ; syslog ]
    ~connect "Logs_syslog_mirage.Udp"
    (pclock @-> stackv4v6 @-> job)

let optional_syslog pclock stack =
  if_impl (Key.value enable_monitoring)
    (syslog $ pclock $ stack)
    noop

let () =
  register "unikernel" [
    optional_syslog default_posix_clock management_stack ;
    main $ <stuff>
  ]
```

TL;DR: the syslog device from mirage is not used; there's a second network device for syslog; this is as well used for mirage-monitoring (counters / stats / dynamic changing the log level).

Of course, the code above could be embedded into mirage, but I don't think that adding code to mirage is good (since I'd prefer it to be more open-world than the closed world assumption; also having it in the config.ml makes the unikernel less magic).
